### PR TITLE
xpra: 4.0.2 -> 4.0.6, add update script

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Motivation for this change
Packages should be kept up-to-date as much as possible. ~~The update script piggybacks of the Debian repository index that upstream maintains in order to determine the latest version.~~

###### Things done
Add script, and use script to update the package.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lourkeur** [comment]
I am unable to run nixpkgs-review at the moment because I don't have access to `jdk-8u261`. (outdated, #108810)

File: pkgs/tools/X11/xpra/update.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl gnused common-updater-scripts
+
+testing=bullseye  # Current Debian testing distribution
```
Comment (SuperSandro2000) [inline pkgs/tools/X11/xpra/update.sh:4]: Can we use sid here?

Comment (lourkeur) [reply pkgs/tools/X11/xpra/update.sh:4]: Yep. They don't seem to have any rolling release distros on there, and I couldn't find a spot on xpra.org that lists the most recent stable version besides this one.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108969` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>run-scaled</li>
  </ul>
</details>

**Comment by lourkeur** [comment]
Changed the update script to analyze the directory listing page instead. @SuperSandro2000 do you think that's preferable?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108969` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>ib-controller: log was empty</li>
  </ul>
</details>
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>run-scaled</li>
<li>xpra</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
> Changed the update script to analyze the directory listing page instead. @SuperSandro2000 do you think that's preferable?

Not sure if it can easily break but it should work for now.

**Comment by lourkeur** [comment]
Ready for full review.

`ib-controller` builds fine if you have the unfree `/nix/store/g51h2dcwsmb5mahx6h6z7bvbj45n7ijy-jdk-8u271-linux-x64.tar.gz` in store.
```
$ nix run nixpkgs#nixpkgs-review pr 108969
[...]

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/108969

3 packages built:
ib-controller run-scaled xpra

**Pull request merged**